app/javascript/mastodon/components/dropdown_menu.js

```diff
@@ -16,6 +16,12 @@ class DropdownMenu extends React.PureComponent {
     ariaLabel: "Menu"
   };
 
+  constructor(props) {
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+  }
```

Comment:
`onClick={this.handleClick}` and then `this` being called inside of `handleClick` I believe is what makes this necessary.